### PR TITLE
Update Getting Started Documentation

### PR DESCRIPTION
**Pull request by MarcusHolly**
## Summary/Motivation:
The Getting Started documentation still recommends python 3.8, which is no longer compatible with IDAES 2.6

## Changes proposed in this PR:
- Changes the default python version from 3.8 to 3.11
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by adam-a-a — APPROVED** [review]

File: docs/getting_started.rst [hunk 1]
```diff
@@ -21,7 +21,7 @@ Create a Conda environment (in this example, named ``watertap``) where WaterTAP
 
 .. code-block:: shell
 
-   conda create --name watertap --yes python=3.8 pip=21.1
+   conda create --name watertap --yes python=3.9 pip=21.1
```
Comment (lbianchi-lbl) [inline docs/getting_started.rst]:
```suggestion
   conda create --name watertap --yes python=3.11
```

File: docs/getting_started.rst [hunk 2]
```diff
@@ -58,7 +58,7 @@ Create a Conda environment (in this example, named ``watertap``) where WaterTAP
...
```
Comment (lbianchi-lbl) [inline docs/getting_started.rst]:
```suggestion
   conda create --name watertap --yes python=3.11
```

File: docs/getting_started.rst [hunk 3]
```diff
@@ -143,7 +143,7 @@ Create a Conda environment (in this example, named ``watertap-dev``) where Water
 
 .. code-block:: shell
 
-   conda create --name watertap-dev --yes python=3.8 pip=21.1 && conda activate watertap-dev
+   conda create --name watertap-dev --yes python=3.9 pip=21.1 && conda activate watertap-dev
```
Comment (lbianchi-lbl) [inline docs/getting_started.rst]:
```suggestion
   conda create --name watertap-dev --yes python=3.11 && conda activate watertap-dev
```

**Review by lbianchi-lbl — COMMENTED** [review]
Thanks for opening this @MarcusHolly. Two changes I'd suggest:

- **Use Python 3.11 instead of 3.9**: mostly because it's the newest version that's tested to fully work with Pyomo/IDAES/WaterTAP. We'll still be testing 3.9 in CI for compatibility but I don't think there's a strong reason not to have a newer version as the default
- **Remove the version constraint on `pip` installed through Conda**: I vaguely remember this being necessary in the past because of some issue/bug, but I don't think it's needed anymore

**Review by lbianchi-lbl — APPROVED** [review]

**Pull request merged**